### PR TITLE
Fix passing of netrc credentials to clone step

### LINKED ISSUE #479: Netrc not passed to clone container
the first step of a pipeline is the close step, if not set by pipeline config it will be added by default.

this environment should have access to Netrc variables:
 https://github.com/woodpecker-ci/woodpecker/blob/3c5827f08a9b42851c118cf1ab1254f68d66f0dd/pipeline/frontend/yaml/compiler/option.go#L71-L87

but that's not the case!

block #473 (-> https://github.com/woodpecker-ci/plugin-git/issues/4)

### PR DESCRIPTION
**Pull request by anbraten**
closes #479

### CODE REVIEW

**Comment by 6543** [comment]
status: could not verify fix by manual testing :/

**Comment by anbraten** [comment]
@6543 I have tested it with the adjusted code (with `woodpeckerci/plugin-git:next`).
Could you verify the repo you test has: `private: false` with `https://ci.example.tld/api/repos/<owner>/<repo>`.

My test config looks like:

```yml
pipeline:
  test:
    image: golang:1.16
    group: test
    commands:
      - env | grep CI_
      - echo "hello world 123"
```

File: pipeline/frontend/yaml/compiler/compiler.go [hunk 1]
```diff
@@ -108,17 +110,19 @@ func (c *Compiler) Compile(conf *yaml.Config) *backend.Config {
...
+			Vargs:       map[string]interface{}{"depth": "0"},
+			Environment: c.cloneEnv,
 		}
+		// // TODO: migrate to woodpeckerci/plugin-git:latest (multi arch)
+		// switch c.metadata.Sys.Arch {
+		// case "linux/arm":
+		// 	container.Image = "plugins/git:linux-arm"
+		// case "linux/arm64":
+		// 	container.Image = "plugins/git:linux-arm64"
+		// }
```
Comment (anbraten) [inline pipeline/frontend/yaml/compiler/compiler.go:125]: Should we just remove this as the plugin is outdated (old CAs) and will be replaced with our own multiarch plugin at some point anyways?

File: pipeline/frontend/yaml/compiler/compiler.go [hunk 2]
```diff
@@ -108,17 +110,19 @@ func (c *Compiler) Compile(conf *yaml.Config) *backend.Config {
...
-			Vargs: map[string]interface{}{"depth": "0"},
-		}
-		// TODO: migrate to woodpeckerci/plugin-git:latest (multi arch)
-		switch c.metadata.Sys.Arch {
-		case "linux/arm":
-			container.Image = "plugins/git:linux-arm"
-		case "linux/arm64":
-			container.Image = "plugins/git:linux-arm64"
+			Name: "clone",
+			// Image:       "woodpeckerci/plugin-git:latest",
```
Comment (anbraten) [inline pipeline/frontend/yaml/compiler/compiler.go:114]: Should we go with `:next` and add a todo and change it just before releasing 0.15.0?

**Review by 6543 — APPROVED** [review]
works

**Pull request merged**